### PR TITLE
Netmap: Backport to 6.0.x

### PR DESCRIPTION
**Pull request by jlucovsky**
Continuation of #6689 

Batch backport of Netmap issues to 6.0.x:
-  [Issue #4582](https://redmine.openinfosecfoundation.org/issues/4852) that provide access to the Netmap API changes in Netmap API versions 14+.
- [Issue #4883](https://redmine.openinfosecfoundation.org/issues/4883) allows specification of Netmap libraries during `configure`

The new APIs are only available when you configure Suricata and include `--enable-netmap-v14`
Specify non-standard location of Netmap libraries with `--with-netmap-libraries=dir`

Updates
- Fix compilation issue when Netmap headers are not available.
-  Added issue 4483

Describe changes:
- Changes to `configure` to choose the new APIs. Use `--enable-netmap-v14` to select the new APIs
- Netmap V14+ api changes (cherry-picked and amended as needed).
- Changes to `configure` to specify Netmap library directory. Use `--with-netmap-libraries=dir`

#suricata-verify-pr:
#suricata-verify-repo:
#suricata-verify-branch:
#suricata-update-pr:
#suricata-update-repo:
#suricata-update-branch:
#libhtp-pr:
#libhtp-repo:
#libhtp-branch:

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information:
 

ERROR: QA failed on tlpw2_single_alerts_cmp.

ERROR: QA failed on tlpw2_autofp_alerts_cmp.

ERROR: QA failed on tlpr1_alerts_cmp.
 | field | test | baseline | % |
|--|--|--|--| 
| | generic_stats_chk |
| .tcp.reassembly_gap | 90932 | 80952 | 112.33% | 

Pipeline 5538

File: src/source-netmap.c [hunk 1]
```diff
@@ -185,12 +170,21 @@ static SCMutex netmap_devlist_lock = SCMUTEX_INITIALIZER;
  */
 int NetmapGetRSSCount(const char *ifname)
```
Comment (victorjulien) [inline src/source-netmap.c:171]: changes like this worry me a bit for backports, can we avoid these for the case where ppl did not specify to use the new code?

**Review by victorjulien — CHANGES_REQUESTED** [review]
I'd like there to be a much cleaner separation between the old and new api support, and/or a reduction of changes overall. I see a lot of things move around for no clear reason, making it hard to track actual changes. Changes for the backport should be as minimal as possible.

**Comment by jlucovsky** [comment]
Continued in #7070

**Pull request closed without merging**